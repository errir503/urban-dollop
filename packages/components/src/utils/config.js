/**
 * Internal dependencies
 */
import CONFIG from './config-values';

/**
 * @param {keyof CONFIG} name The variable name
 * @return {string | number} The variable
 */
export const config = ( name ) => CONFIG[ name ];
