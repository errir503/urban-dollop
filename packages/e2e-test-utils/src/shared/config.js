const WP_ADMIN_USER = {
	username: 'admin',
	password: 'password',
};

const {
	WP_USERNAME = WP_ADMIN_USER.username,
	WP_PASSWORD = WP_ADMIN_USER.password,
	WP_BASE_URL = 'http://localhost:8889',
} = process.env;

export { WP_ADMIN_USER, WP_USERNAME, WP_PASSWORD, WP_BASE_URL };
