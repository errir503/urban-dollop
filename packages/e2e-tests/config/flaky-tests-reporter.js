/**
 * A **flaky** test is defined as a test which passed after auto-retrying.
 * - By default, all tests run once if they pass.
 * - If a test fails, it will automatically re-run at most 2 times.
 * - If it pass after retrying (below 2 times), then it's marked as **flaky**
 *   but displayed as **passed** in the original test suite.
 * - If it fail all 3 times, then it's a **failed** test.
 */
/**
 * External dependencies
 */
const fs = require( 'fs' ).promises;
const path = require( 'path' );
const { formatResultsErrors } = require( 'jest-message-util' );
const filenamify = require( 'filenamify' );

class FlakyTestsReporter {
	constructor( globalConfig, options ) {
		this._globalConfig = globalConfig;
		this._options = options;

		this.failingTestCaseResults = new Map();
	}

	async onRunStart() {
		try {
			fs.mkdir( 'flaky-tests' );
		} catch ( err ) {
			// Ignore the error if the directory already exists.
			if ( err.code !== 'EEXIST' ) {
				throw err;
			}
		}
	}

	async onTestCaseResult( test, testCaseResult ) {
		const testPath = path.relative( this._globalConfig.rootDir, test.path );
		const testTitle = testCaseResult.title;

		switch ( testCaseResult.status ) {
			case 'failed': {
				if ( ! this.failingTestCaseResults.has( testTitle ) ) {
					this.failingTestCaseResults.set( testTitle, [] );
				}
				this.failingTestCaseResults
					.get( testTitle )
					.push( testCaseResult );
				break;
			}
			case 'passed': {
				if ( this.failingTestCaseResults.has( testTitle ) ) {
					const failingResults =
						this.failingTestCaseResults.get( testTitle );

					await fs.writeFile(
						`flaky-tests/${ filenamify( testTitle ) }.json`,
						JSON.stringify( {
							version: 1,
							runner: 'jest-circus',
							title: testTitle,
							path: testPath,
							results: failingResults,
						} ),
						'utf-8'
					);

					// Don't silence flaky error messages for debugging reason.
					// eslint-disable-next-line no-console
					console.error(
						`Test passed after ${ failingResults.length } failed ${
							failingResults.length === 1 ? 'attempt' : 'attempts'
						}:`
					);
					// eslint-disable-next-line no-console
					console.error(
						formatResultsErrors(
							failingResults,
							this._globalConfig,
							{},
							test.path
						)
					);
				}
				break;
			}
			default:
				break;
		}
	}

	onRunComplete() {
		this.failingTestCaseResults.clear();
	}
}

module.exports = FlakyTestsReporter;
