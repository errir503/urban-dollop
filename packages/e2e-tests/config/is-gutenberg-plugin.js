global.process.env = {
	...global.process.env,
	// Inject the `IS_GUTENBERG_PLUGIN` global, used for feature flagging.
	// eslint-disable-next-line @wordpress/is-gutenberg-plugin
	IS_GUTENBERG_PLUGIN: process.env.npm_package_config_IS_GUTENBERG_PLUGIN,
};
