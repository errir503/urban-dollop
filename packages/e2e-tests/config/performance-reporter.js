/**
 * External dependencies
 */
const { readFileSync, existsSync } = require( 'fs' );
const path = require( 'path' );
const chalk = require( 'chalk' );

function average( array ) {
	return array.reduce( ( a, b ) => a + b ) / array.length;
}

function round( number, decimalPlaces = 2 ) {
	const factor = Math.pow( 10, decimalPlaces );
	return Math.round( number * factor ) / factor;
}

const title = chalk.bold;
const success = chalk.bold.green;

class PerformanceReporter {
	onTestResult( test ) {
		const dirname = path.dirname( test.path );
		const basename = path.basename( test.path, '.js' );
		const filepath = path.join( dirname, basename + '.results.json' );

		if ( ! existsSync( filepath ) ) {
			return;
		}

		const results = readFileSync( filepath, 'utf8' );
		const {
			serverResponse,
			firstPaint,
			domContentLoaded,
			loaded,
			firstContentfulPaint,
			firstBlock,
			type,
			focus,
			listViewOpen,
			inserterOpen,
			inserterHover,
			inserterSearch,
		} = JSON.parse( results );

		if ( serverResponse && serverResponse.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Loading Time:' ) }
Average time to server response (subtracted from client side metrics): ${ success(
				round( average( serverResponse ) ) + 'ms'
			) }
Average time to first paint: ${ success(
				round( average( firstPaint ) ) + 'ms'
			) }
Average time to DOM content load: ${ success(
				round( average( domContentLoaded ) ) + 'ms'
			) }
Average time to load: ${ success( round( average( loaded ) ) + 'ms' ) }
Average time to first contentful paint: ${ success(
				round( average( firstContentfulPaint ) ) + 'ms'
			) }
Average time to first block: ${ success(
				round( average( firstBlock ) ) + 'ms'
			) }` );
		}

		if ( type && type.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Typing Performance:' ) }
Average time to type character: ${ success( round( average( type ) ) + 'ms' ) }
Slowest time to type character: ${ success(
				round( Math.max( ...type ) ) + 'ms'
			) }
Fastest time to type character: ${ success(
				round( Math.min( ...type ) ) + 'ms'
			) }` );
		}

		if ( focus && focus.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Block Selection Performance:' ) }
Average time to select a block: ${ success( round( average( focus ) ) + 'ms' ) }
Slowest time to select a block: ${ success(
				round( Math.max( ...focus ) ) + 'ms'
			) }
Fastest time to select a block: ${ success(
				round( Math.min( ...focus ) ) + 'ms'
			) }` );
		}

		if ( listViewOpen && listViewOpen.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Opening List View Performance:' ) }
Average time to open list view: ${ success(
				round( average( listViewOpen ) ) + 'ms'
			) }
Slowest time to open list view: ${ success(
				round( Math.max( ...listViewOpen ) ) + 'ms'
			) }
Fastest time to open list view: ${ success(
				round( Math.min( ...listViewOpen ) ) + 'ms'
			) }` );
		}

		if ( inserterOpen && inserterOpen.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Opening Global Inserter Performance:' ) }
Average time to open global inserter: ${ success(
				round( average( inserterOpen ) ) + 'ms'
			) }
Slowest time to open global inserter: ${ success(
				round( Math.max( ...inserterOpen ) ) + 'ms'
			) }
Fastest time to open global inserter: ${ success(
				round( Math.min( ...inserterOpen ) ) + 'ms'
			) }` );
		}

		if ( inserterSearch && inserterSearch.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Inserter Search Performance:' ) }
Average time to type the inserter search input: ${ success(
				round( average( inserterSearch ) ) + 'ms'
			) }
Slowest time to type the inserter search input: ${ success(
				round( Math.max( ...inserterSearch ) ) + 'ms'
			) }
Fastest time to type the inserter search input: ${ success(
				round( Math.min( ...inserterSearch ) ) + 'ms'
			) }` );
		}

		if ( inserterHover && inserterHover.length ) {
			// eslint-disable-next-line no-console
			console.log( `
${ title( 'Inserter Block Item Hover Performance:' ) }
Average time to move mouse between two block item in the inserter: ${ success(
				round( average( inserterHover ) ) + 'ms'
			) }
Slowest time to move mouse between two block item in the inserter: ${ success(
				round( Math.max( ...inserterHover ) ) + 'ms'
			) }
Fastest time to move mouse between two block item in the inserter: ${ success(
				round( Math.min( ...inserterHover ) ) + 'ms'
			) }` );
		}

		// eslint-disable-next-line no-console
		console.log( '' );
	}
}

module.exports = PerformanceReporter;
