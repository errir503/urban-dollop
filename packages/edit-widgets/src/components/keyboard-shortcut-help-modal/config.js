/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const textFormattingShortcuts = [
	{
		keyCombination: { modifier: 'primary', character: 'b' },
		description: __( 'Make the selected text bold.' ),
	},
	{
		keyCombination: { modifier: 'primary', character: 'i' },
		description: __( 'Make the selected text italic.' ),
	},
	{
		keyCombination: { modifier: 'primary', character: 'k' },
		description: __( 'Convert the selected text into a link.' ),
	},
	{
		keyCombination: { modifier: 'primaryShift', character: 'k' },
		description: __( 'Remove a link.' ),
	},
	{
		keyCombination: { modifier: 'primary', character: 'u' },
		description: __( 'Underline the selected text.' ),
	},
];
