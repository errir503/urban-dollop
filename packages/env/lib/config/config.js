'use strict';
/**
 * External dependencies
 */
const fs = require( 'fs' ).promises;
const path = require( 'path' );
const os = require( 'os' );

/**
 * Internal dependencies
 */
const detectDirectoryType = require( './detect-directory-type' );
const { validateConfig, ValidationError } = require( './validate-config' );
const readRawConfigFile = require( './read-raw-config-file' );
const parseConfig = require( './parse-config' );
const { includeTestsPath, parseSourceString } = parseConfig;
const md5 = require( '../md5' );

/**
 * wp-env configuration.
 *
 * @typedef WPConfig
 * @property {string}                           name                    Name of the environment.
 * @property {string}                           configDirectoryPath     Path to the .wp-env.json file.
 * @property {string}                           workDirectoryPath       Path to the work directory located in ~/.wp-env.
 * @property {string}                           dockerComposeConfigPath Path to the docker-compose.yml file.
 * @property {boolean}                          detectedLocalConfig     If true, wp-env detected local config and used it.
 * @property {Object.<string, WPServiceConfig>} env                     Specific config for different environments.
 * @property {boolean}                          debug                   True if debug mode is enabled.
 */

/**
 * Base-level config for any particular environment. (development/tests/etc)
 *
 * @typedef WPServiceConfig
 * @property {WPSource}                  coreSource    The WordPress installation to load in the environment.
 * @property {WPSource[]}                pluginSources Plugins to load in the environment.
 * @property {WPSource[]}                themeSources  Themes to load in the environment.
 * @property {number}                    port          The port to use.
 * @property {Object}                    config        Mapping of wp-config.php constants to their desired values.
 * @property {Object.<string, WPSource>} mappings      Mapping of WordPress directories to local directories which should be mounted.
 * @property {string}                    phpVersion    Version of PHP to use in the environments, of the format 0.0.
 */

/**
 * A WordPress installation, plugin or theme to be loaded into the environment.
 *
 * @typedef WPSource
 * @property {'local'|'git'|'zip'} type     The source type.
 * @property {string}              path     The path to the WordPress installation, plugin or theme.
 * @property {?string}             url      The URL to the source download if the source type is not local.
 * @property {?string}             ref      The git ref for the source if the source type is 'git'.
 * @property {string}              basename Name that identifies the WordPress installation, plugin or theme.
 */

/**
 * Reads, parses, and validates the given .wp-env.json file into a wp-env config
 * object for internal use.
 *
 * @param {string} configPath Path to the .wp-env.json file.
 *
 * @return {WPConfig} A parsed and validated wp-env config object.
 */
module.exports = async function readConfig( configPath ) {
	const configDirectoryPath = path.dirname( configPath );
	const workDirectoryPath = path.resolve(
		await getHomeDirectory(),
		md5( configPath )
	);

	// The specified base configuration from .wp-env.json or from the local
	// source type which was automatically detected.
	const baseConfig =
		( await readRawConfigFile( '.wp-env.json', configPath ) ) ||
		( await getDefaultBaseConfig( configPath ) );

	// Overriden .wp-env.json on a per-user case.
	const overrideConfig =
		( await readRawConfigFile(
			'.wp-env.override.json',
			configPath.replace( /\.wp-env\.json$/, '.wp-env.override.json' )
		) ) || {};

	const detectedLocalConfig =
		Object.keys( { ...baseConfig, ...overrideConfig } ).length > 0;

	// Default configuration which is overridden by .wp-env.json files.
	const defaultConfiguration = {
		core: null, // Indicates that the latest stable version should ultimately be used.
		phpVersion: null,
		plugins: [],
		themes: [],
		port: 8888,
		mappings: {},
		config: {
			WP_DEBUG: true,
			SCRIPT_DEBUG: true,
			WP_ENVIRONMENT_TYPE: 'local',
			WP_PHP_BINARY: 'php',
			WP_TESTS_EMAIL: 'admin@example.org',
			WP_TESTS_TITLE: 'Test Blog',
			WP_TESTS_DOMAIN: 'localhost',
			WP_SITEURL: 'http://localhost',
			WP_HOME: 'http://localhost',
		},
		env: {
			development: {}, // No overrides needed, but it should exist.
			tests: {
				config: { WP_DEBUG: false, SCRIPT_DEBUG: false },
				port: 8889,
			},
		},
	};

	// A quick validation before merging on a service by service level allows us
	// to check the root configuration options and provide more helpful errors.
	validateConfig(
		mergeWpServiceConfigs( [
			defaultConfiguration,
			baseConfig,
			overrideConfig,
		] )
	);

	// A unique array of the environments specified in the config options.
	// Needed so that we can override settings per-environment, rather than
	// overwriting each environment key.
	const getEnvKeys = ( config ) => Object.keys( config.env || {} );
	const allEnvs = [
		...new Set( [
			...getEnvKeys( defaultConfiguration ),
			...getEnvKeys( baseConfig ),
			...getEnvKeys( overrideConfig ),
		] ),
	];

	// Returns a pair with the root config options and the specific environment config options.
	const getEnvConfig = ( config, envName ) => [
		config,
		config.env && config.env[ envName ] ? config.env[ envName ] : {},
	];

	// Merge each of the specified environment-level overrides.
	const allPorts = new Set(); // Keep track of unique ports for validation.
	const env = {};
	for ( const envName of allEnvs ) {
		env[ envName ] = await parseConfig(
			validateConfig(
				mergeWpServiceConfigs( [
					...getEnvConfig( defaultConfiguration, envName ),
					...getEnvConfig( baseConfig, envName ),
					...getEnvConfig( overrideConfig, envName ),
				] ),
				envName
			),
			{ workDirectoryPath }
		);
		allPorts.add( env[ envName ].port );
	}

	if ( allPorts.size !== allEnvs.length ) {
		throw new ValidationError(
			'Invalid .wp-env.json: Each port value must be unique.'
		);
	}

	return withOverrides( {
		name: path.basename( configDirectoryPath ),
		dockerComposeConfigPath: path.resolve(
			workDirectoryPath,
			'docker-compose.yml'
		),
		configDirectoryPath,
		workDirectoryPath,
		detectedLocalConfig,
		env,
	} );
};

/**
 * Deep-merges the values in the given service environment. This allows us to
 * merge the wp-config.php values instead of overwriting them. Note that this
 * merges configs before they have been validated, so the passed config shape
 * will not match the WPServiceConfig type.
 *
 * @param {Object[]} configs Array of raw service config objects to merge.
 *
 * @return {Object} The merged configuration object.
 */
function mergeWpServiceConfigs( configs ) {
	// Returns an array of nested values in the config object. For example,
	// an array of all the wp-config objects.
	const mergeNestedObjs = ( key ) =>
		Object.assign(
			{},
			...configs.map( ( config ) => {
				if ( ! config[ key ] ) {
					return {};
				} else if ( typeof config[ key ] === 'object' ) {
					return config[ key ];
				}
				throw new ValidationError(
					`Invalid .wp-env.json: "${ key }" must be an object.`
				);
			} )
		);

	const mergedConfig = {
		...Object.assign( {}, ...configs ),
		config: mergeNestedObjs( 'config' ),
		mappings: mergeNestedObjs( 'mappings' ),
	};

	delete mergedConfig.env;
	return mergedConfig;
}

/**
 * Detects basic config options to use if the .wp-env.json config file does not
 * exist. For example, if the local directory contains a plugin, that will be
 * added to the default plugin sources.
 *
 * @param {string} configPath A path to the config file for the source to detect.
 * @return {Object} Basic config options for the detected source type. Empty
 *                  object if no config detected.
 */
async function getDefaultBaseConfig( configPath ) {
	const configDirectoryPath = path.dirname( configPath );
	const type = await detectDirectoryType( configDirectoryPath );

	if ( type === 'core' ) {
		return { core: '.' };
	} else if ( type === 'plugin' ) {
		return { plugins: [ '.' ] };
	} else if ( type === 'theme' ) {
		return { themes: [ '.' ] };
	}

	return {};
}

/**
 * Overrides keys in the config object with set environment variables or options
 * which should be merged.
 *
 * @param {WPConfig} config fully parsed configuration object.
 * @return {WPConfig} configuration object with overrides applied.
 */
function withOverrides( config ) {
	const workDirectoryPath = config.workDirectoryPath;
	// Override port numbers with environment variables.
	config.env.development.port =
		getNumberFromEnvVariable( 'WP_ENV_PORT' ) ||
		config.env.development.port;
	config.env.tests.port =
		getNumberFromEnvVariable( 'WP_ENV_TESTS_PORT' ) ||
		config.env.tests.port;

	// Override WordPress core with environment variable.
	if ( process.env.WP_ENV_CORE ) {
		const coreSource = includeTestsPath(
			parseSourceString( process.env.WP_ENV_CORE, { workDirectoryPath } ),
			{ workDirectoryPath }
		);
		config.env.development.coreSource = coreSource;
		config.env.tests.coreSource = coreSource;
	}

	// Override PHP version with environment variable.
	config.env.development.phpVersion =
		process.env.WP_ENV_PHP_VERSION || config.env.development.phpVersion;
	config.env.tests.phpVersion =
		process.env.WP_ENV_PHP_VERSION || config.env.tests.phpVersion;

	const updateEnvUrl = ( configKey ) => {
		[ 'development', 'tests' ].forEach( ( envKey ) => {
			try {
				const baseUrl = new URL(
					config.env[ envKey ].config[ configKey ]
				);

				// Don't overwrite the port of WP_HOME when set.
				if ( ! ( configKey === 'WP_HOME' && !! baseUrl.port ) ) {
					baseUrl.port = config.env[ envKey ].port;
				}

				config.env[ envKey ].config[ configKey ] = baseUrl.toString();
			} catch ( error ) {
				throw new ValidationError(
					`Invalid .wp-env.json: config.${ configKey } must be a valid URL.`
				);
			}
		} );
	};

	// Update wp config options to include the correct port number in the URL.
	updateEnvUrl( 'WP_SITEURL' );
	updateEnvUrl( 'WP_HOME' );

	return config;
}

/**
 * Parses an environment variable which should be a number.
 *
 * Throws an error if the variable cannot be parsed to a number.
 * Returns null if the environment variable has not been specified.
 *
 * @param {string} varName The environment variable to check (e.g. WP_ENV_PORT).
 * @return {null|number} The number. Null if it does not exist.
 */
function getNumberFromEnvVariable( varName ) {
	// Allow use of the default if it does not exist.
	if ( ! process.env[ varName ] ) {
		return null;
	}

	const maybeNumber = parseInt( process.env[ varName ] );

	// Throw an error if it is not parseable as a number.
	if ( isNaN( maybeNumber ) ) {
		throw new ValidationError(
			`Invalid environment variable: ${ varName } must be a number.`
		);
	}

	return maybeNumber;
}

/**
 * Gets the `wp-env` home directory in which generated files are created.
 *
 * By default: '~/.wp-env/'. On Linux with snap packages: '~/wp-env/'. Can be
 * overridden with the WP_ENV_HOME environment variable.
 *
 * @return {Promise<string>} The absolute path to the `wp-env` home directory.
 */
async function getHomeDirectory() {
	// Allow user to override download location.
	if ( process.env.WP_ENV_HOME ) {
		return path.resolve( process.env.WP_ENV_HOME );
	}

	/**
	 * Installing docker with Snap Packages on Linux is common, but does not
	 * support hidden directories. Therefore we use a public directory when
	 * snap packages exist.
	 *
	 * @see https://github.com/WordPress/gutenberg/issues/20180#issuecomment-587046325
	 */
	return path.resolve(
		os.homedir(),
		!! ( await fs.stat( '/snap' ).catch( () => false ) )
			? 'wp-env'
			: '.wp-env'
	);
}
