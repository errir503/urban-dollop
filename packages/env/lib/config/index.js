/**
 * Internal dependencies
 */
const readConfig = require( './config' );
const { ValidationError } = require( './validate-config' );
const dbEnv = require( './db-env' );

/**
 * @typedef {import('./config').WPConfig} WPConfig
 * @typedef {import('./config').WPServiceConfig} WPServiceConfig
 * @typedef {import('./config').WPSource} WPSource
 */

module.exports = {
	ValidationError,
	readConfig,
	dbEnv,
};
