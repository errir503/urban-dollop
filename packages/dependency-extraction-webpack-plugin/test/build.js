/**
 * External dependencies
 */
const fs = require( 'fs' );
const glob = require( 'glob' ).sync;
const mkdirp = require( 'mkdirp' ).sync;
const path = require( 'path' );
const rimraf = require( 'rimraf' ).sync;
const webpack = require( 'webpack' );

const fixturesPath = path.join( __dirname, 'fixtures' );
const configFixtures = fs.readdirSync( fixturesPath ).sort();

afterAll( () => rimraf( path.join( __dirname, 'build' ) ) );

describe.each( configFixtures )( 'Webpack `%s`', ( configCase ) => {
	const testDirectory = path.join( fixturesPath, configCase );
	const outputDirectory = path.join( __dirname, 'build', configCase );

	beforeEach( () => {
		rimraf( outputDirectory );
		mkdirp( outputDirectory );
	} );

	// This afterEach is necessary to prevent watched tests from retriggering on every run.
	afterEach( () => rimraf( outputDirectory ) );

	test( 'should produce expected output', () =>
		new Promise( ( resolve ) => {
			const options = Object.assign(
				{
					context: testDirectory,
					entry: './index.js',
					mode: 'production',
					optimization: {
						minimize: false,
						chunkIds: 'named',
						moduleIds: 'named',
					},
					output: {},
				},
				require( path.join( testDirectory, 'webpack.config.js' ) )
			);
			options.output.path = outputDirectory;

			webpack( options, ( err, stats ) => {
				expect( err ).toBeNull();

				const assetFiles = glob(
					`${ outputDirectory }/+(*.asset|assets).@(json|php)`
				);
				const hasCombinedAssets = ( options.plugins || [] ).some(
					( plugin ) => !! ( plugin.options || {} ).combineAssets
				);
				const entrypointCount =
					typeof options.entry === 'object'
						? Object.keys( options.entry ).length
						: 1;
				const expectedLength = hasCombinedAssets ? 1 : entrypointCount;
				expect( assetFiles ).toHaveLength( expectedLength );

				// Asset files should match.
				assetFiles.forEach( ( assetFile ) => {
					expect(
						fs.readFileSync( assetFile, 'utf-8' )
					).toMatchSnapshot( 'Asset file should match snapshot' );
				} );

				const compareByModuleIdentifier = ( m1, m2 ) => {
					const i1 = m1.identifier();
					const i2 = m2.identifier();
					if ( i1 < i2 ) return -1;
					if ( i1 > i2 ) return 1;
					return 0;
				};

				// Webpack stats external modules should match.
				const externalModules = stats.compilation.modules
					.filter( ( { externalType } ) => externalType )
					.sort( compareByModuleIdentifier )
					.map( ( module ) => ( {
						externalType: module.externalType,
						request: module.request,
						userRequest: module.userRequest,
					} ) );
				expect( externalModules ).toMatchSnapshot(
					'External modules should match snapshot'
				);

				resolve();
			} );
		} ) );
} );
