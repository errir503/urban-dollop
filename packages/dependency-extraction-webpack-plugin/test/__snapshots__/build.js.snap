// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack \`combine-assets\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('fileA.js' => array('dependencies' => array('lodash', 'wp-blob'), 'version' => 'a3da22ced6876ec052d2861d383960fc'), 'fileB.js' => array('dependencies' => array('wp-token-list'), 'version' => '1a2b3802cc39dc1a607ecc4d0b23fcb0'));"`;

exports[`Webpack \`combine-assets\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "tokenList",
    ],
    "userRequest": "@wordpress/token-list",
  },
]
`;

exports[`Webpack \`dynamic-import\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => 'd5c613274c346f46e16ab4d320fc01e6');"`;

exports[`Webpack \`dynamic-import\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`Webpack \`function-output-filename\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => 'f6dc46e27c3a9e7338ed4fa9fdf8f606');"`;

exports[`Webpack \`function-output-filename\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`Webpack \`no-default\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array(), 'version' => '0a2f4c4cfbc33dde065a2e4be1e6337c');"`;

exports[`Webpack \`no-default\` should produce expected output: External modules should match snapshot 1`] = `Array []`;

exports[`Webpack \`no-deps\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array(), 'version' => 'e8e65570823ee9206bdada3a0f11b610');"`;

exports[`Webpack \`no-deps\` should produce expected output: External modules should match snapshot 1`] = `Array []`;

exports[`Webpack \`output-format-json\` should produce expected output: Asset file should match snapshot 1`] = `"{\\"dependencies\\":[\\"lodash\\"],\\"version\\":\\"aeb5066a5e17aea01932c77baf342372\\"}"`;

exports[`Webpack \`output-format-json\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
]
`;

exports[`Webpack \`overrides\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('wp-blob', 'wp-script-handle-for-rxjs', 'wp-url'), 'version' => '67d711ce8940ddb82e7f264f61a5a3d9');"`;

exports[`Webpack \`overrides\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "rxjs",
    "userRequest": "rxjs",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "rxjs",
      "operators",
    ],
    "userRequest": "rxjs/operators",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "url",
    ],
    "userRequest": "@wordpress/url",
  },
]
`;

exports[`Webpack \`wordpress\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '202b3ce17cfd72799ce45e7efa04d83c');"`;

exports[`Webpack \`wordpress\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;

exports[`Webpack \`wordpress-require\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '47b0c1540a1caf08b8931e4a4328db04');"`;

exports[`Webpack \`wordpress-require\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "window",
    "request": "lodash",
    "userRequest": "lodash",
  },
  Object {
    "externalType": "window",
    "request": Array [
      "wp",
      "blob",
    ],
    "userRequest": "@wordpress/blob",
  },
]
`;
